dts/bindings/kscan/zephyr,kscan-input.yaml

```diff
@@ -0,0 +1,23 @@
+# Copyright 2023 Google LLC
+# SPDX-License-Identifier: Apache-2.0
+
+description: |
+  Input to kscan adapter.
+
+  Allows using an input device with the kscan API. Define as a child node of
+  the input device, for example
+
+  chosen {
+    zephyr,keyboard-scan = &kscan_input;
+  };
+
+  ft5336@38 {
+    ...
+    kscan_input: kscan-input {
+      compatible = "zephyr,kscan-input";
+    };
```

Comment:
@galak debating on whether to use this and `DT_INST_PARENT()` (as proposed here) or have a separate node with an `input` phandle property. I like the child node approach, seems more clear to me, thinking about using it for defining and LVGL "sink" as well and use it to get few options that are currently in Kconfig but really depend on the board (`swap-xy`, `invert-x`, `invert-y`...). What do you think?